recipes/default.rb

```diff
@@ -15,14 +15,14 @@
   node['rackspace_users']['data_bag'],
   node['rackspace_users']['data_bag_item']).to_hash.select { |user, user_data| user != 'id' }
 
-node_tags = node['rackspace_users']['node_tags']
+node_groups = node['rackspace_users']['node_groups']
 
 groups = {}
 
 users.each do |username, user_data|
-  ## The action (if defined in the data bag) should be overwritten to 'remove' if the user's node tags and node's tags don't intersect.
-  ## If the user doesn't declare any tags then it is assumed that is present on all nodes.
-  user_data['action'] = 'remove' if user_data['node_tags'] && (user_data['node_tags'] & node_tags).empty?
+  ## The action (if defined in the data bag) should be overwritten to 'remove' if not all the user's node_groups are in node's node_groups.
+  ## If the user doesn't declare node_groups then it is assumed that is present on all nodes.
+  user_data['action'] = 'remove' if user_data['node_groups'] && !(user_data['node_groups'] - node_groups).empty?
```

Comment:
> before, to create a user at least one of the user’s node_groups should be in the node’s node_groups

To summarize the previous behavior:
- if no groups are assigned to a user, the user will be created on the node
- if groups are assigned to a user, and there is an identical group assigned to the node, the user will be created on the node
- if groups are assigned to a user, but no identical group is assigned to the node, the user will be removed from the node (if present)


> now, I changed it so all of the user’s node_groups have to be in the node’s node_groups to create a user 
To summarize the new behavior being introduced

- if no groups are assigned to a user, the user will be created on the node
- if groups are assigned to a user, and the identical set of groups are assigned to the node, the user will be created on the node
- if groups are assigned to a user, but the identical groups are NOT assigned to the node, the user will be removed from the node (if present)

I'm a bit torn on this one. I use the former functionality more often than I need the latter. I feel like it's especially confusing/unintuitive to remove the user if some groups match and some don't.

I feel like the use case of 'any matching group' is going to be (a) much more intuitive for users of this cookbook and (b) a more common need than 'all groups must match.' I'd love for others to weigh in on this, though. I feel like the functionality is equivalent before and after, just with more complicated logic if we make this change. Perhaps we should default to the simpler logic, and offer the option that they must all match (defaulted to false)? Or does that complicate matters worse?